File: src/android/com/notifications/plugin/Notifications.java
package com.notifications.plugin;

import org.apache.cordova.CordovaPlugin;
import org.apache.cordova.CallbackContext;
import org.json.JSONArray;
import org.json.JSONObject;
import org.json.JSONException;
import android.content.Context;
import android.widget.Toast;

import androidx.annotation.NonNull;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.*;
import com.google.firebase.messaging.*;

import static android.provider.Settings.Global.getString;

public class Notifications extends CordovaPlugin {
  @Override
  public boolean execute(
    String action, 
    JSONArray args, 
    CallbackContext callbackContext
  ) throws JSONException {
    if ("registration".equals(action)) {
      FirebaseMessaging.getInstance().getToken().addOnCompleteListener(new OnCompleteListener<String>() {
        @Override
        public void onComplete(@NonNull Task<String> task) {
          System.out.println(task.getResult());
          callbackContext.success(task.getResult());
        }
      });
      return true;
    } else {
      callbackContext.error("Permission denied");
    }
    return false;
  }
}

File: src/android/Notifications.kt
package notifications

import android.content.Intent
import android.util.Log
import com.google.android.gms.tasks.OnCompleteListener
import com.google.firebase.messaging.FirebaseMessaging
import org.apache.cordova.CallbackContext
import org.apache.cordova.CordovaPlugin
import org.json.JSONArray
import org.json.JSONException


class Notifications: CordovaPlugin() {
  companion object {
    private const val TAG = "pushNotification"
  }

  @Throws(JSONException::class)
  override fun execute(action: String, data: JSONArray, callbackContext: CallbackContext): Boolean {
    var result = true
    try {
      when (action) {
        "registration" -> {
          Log.d(TAG, "registration")

          cordova.threadPool.execute {
            getFirebaseToken(callbackContext)
          }
        }
        "tapped" -> {
          Log.d(TAG, "tapped")

          val activity = cordova.activity
          val extraText = activity.intent.getStringExtra(Intent.EXTRA_TEXT) ?: ""

          cordova.threadPool.execute {
            callbackContext.success(extraText)
          }
        }
        else -> {
          handleError("Invalid action", callbackContext)
          result = false
        }
      }
    } catch (e: Exception) {
      handleError(e.toString(), callbackContext)
      result = false
    }

    return result
  }

  private fun getFirebaseToken(context: CallbackContext) {
    FirebaseMessaging.getInstance().token.addOnCompleteListener(OnCompleteListener { task ->
      if (!task.isSuccessful) {
        Log.w(TAG, "Fetching FCM registration token failed", task.exception)
        return@OnCompleteListener
      }

      val token = task.result
      Log.d(TAG, token)
      context.success(token)
    })
  }

  private fun handleError(errorMsg: String, context: CallbackContext) {
    try {
      Log.e(TAG, errorMsg)
      context.error(errorMsg)
    } catch (e: Exception) {
      Log.e(TAG, e.toString())
    }
  }
}
